src/Kiota.Builder/Writers/CSharp/CSharpConventionService.cs

```diff
@@ -2,49 +2,74 @@
+        private const string PathSegmentPropertyName = "PathSegment";
+        private const string CurrentPathPropertyName = "CurrentPath";
+        private const string HttpCorePropertyName = "HttpCore";
+        public HashSet<string> NullableTypes { get; } = new(StringComparer.OrdinalIgnoreCase) { "int", "bool", "float", "double", "decimal", "long", "Guid", "DateTimeOffset" };
+        public static readonly char NullableMarker = '?';
+        public static string NullableMarkerAsString => "?";
         public override string ParseNodeInterfaceName => "IParseNode";
-        public override string RawUrlPropertyName => "IsRawUrl";
         public override void WriteShortDescription(string description, LanguageWriter writer) {
             if(!string.IsNullOrEmpty(description))
                 writer.WriteLine($"{DocCommentPrefix}<summary>{description}</summary>");
         }
         public override string GetAccessModifier(AccessModifier access)
         {
-            return (access) switch {
-                (AccessModifier.Public) => "public",
-                (AccessModifier.Protected) => "protected",
+            return access switch {
+                AccessModifier.Public => "public",
+                AccessModifier.Protected => "protected",
                 _ => "private",
             };
         }
-        internal void AddRequestBuilderBody(bool addCurrentPath, string returnType, LanguageWriter writer, string suffix = default, string prefix = default) {
+        #pragma warning disable CA1822 // Method should be static
+        internal void AddRequestBuilderBody(bool addCurrentPath, string returnType, LanguageWriter writer, string suffix = default, string prefix = default, IEnumerable<CodeParameter> pathParameters = default) {
             var currentPath = addCurrentPath ? $"{CurrentPathPropertyName} + " : string.Empty;
-            writer.WriteLine($"{prefix}new {returnType}({currentPath}{PathSegmentPropertyName} {suffix}, {HttpCorePropertyName}, false);");
+            var pathParametersSuffix = !(pathParameters?.Any() ?? false) ? string.Empty : $"{string.Join(", ", pathParameters.Select(x => $"{x.Name}"))}, ";
+            writer.WriteLine($"{prefix}new {returnType}({currentPath}{PathSegmentPropertyName} {suffix}, {HttpCorePropertyName}, {pathParametersSuffix}false);");
         }
+        #pragma warning restore CA1822 // Method should be static
         internal bool ShouldTypeHaveNullableMarker(CodeTypeBase propType, string propTypeName) {
             return propType.IsNullable && (NullableTypes.Contains(propTypeName) || (propType is CodeType codeType && codeType.TypeDefinition is CodeEnum));
         }
-        public override string GetTypeString(CodeTypeBase code)
+        private static HashSet<string> _reservedNames;
+        private static readonly object _reservedNamesLock = new();
+        private static HashSet<string> GetReservedNames(CodeElement currentElement) {
```

Comment:
If I understand this correctly, this is more related to namespace names.
We should probably rename this function to reflect that the reserved Names in questions are related to the namespace names and does not really relate to the language reserved keyword.

Reply:
fixed, this should be explicit.